src/Analyzer.JsonRuleEngine/Schema/RuleDefinition.cs

```diff
@@ -39,5 +40,11 @@ internal class RuleDefinition
         /// </summary>
         [JsonProperty(PropertyName = "evaluation")]
         public ExpressionDefinition ExpressionDefinition { get; set; }
+
+        /// <summary>
+        /// Gets or sets an <see cref="Expression"/> that can evaluate a template against this rule.
+        /// </summary>
+        [JsonIgnore]
+        internal Expression Expression { get; set; }
```

Comment:
With this, I think we can now easily implement the more informative messages when rules are violated.